tests/fixtures/local_fixtures.py

```diff
@@ -0,0 +1,156 @@
+        await storage.create_bucket(bucket_id=CONFIG.inbox_bucket)
+    if await storage.does_object_exist(
+        bucket_id=CONFIG.inbox_bucket, object_id=CONFIG.object_id
+    ):
+        await storage.delete_object(
+            bucket_id=CONFIG.inbox_bucket, object_id=CONFIG.object_id
+        )
+    yield storage
+
+
+@pytest_asyncio.fixture
+async def populated_bucket_fixture(
+    local_s3_fixture: S3ObjectStorage,
+) -> event_schemas.FileUploadReceived:
+    """Generate test file and prepare event"""
+    file_size = 20 * 1024**2
+
+    num_parts = math.ceil(file_size / PART_SIZE)
+    with big_temp_file(size=file_size) as random_data:
+
+        checksum = hashlib.sha256(random_data.read()).hexdigest()
+        # rewind file
+        random_data.seek(0)
+
+        upload_id = await local_s3_fixture.init_multipart_upload(
+            bucket_id=CONFIG.inbox_bucket, object_id=CONFIG.object_id
+        )
+        try:
+            for part_number in range(1, num_parts + 1):
+
+                part = random_data.read(PART_SIZE)
+                upload_url = await local_s3_fixture.get_part_upload_url(
+                    upload_id=upload_id,
+                    bucket_id=CONFIG.inbox_bucket,
+                    object_id=CONFIG.object_id,
+                    part_number=part_number,
+                )
+                response = requests.put(url=upload_url, data=part, timeout=2)
+                assert response.status_code == 200
+
+            await local_s3_fixture.complete_multipart_upload(
+                upload_id=upload_id,
+                bucket_id=CONFIG.inbox_bucket,
+                object_id=CONFIG.object_id,
+            )
+
+        except (Exception, KeyboardInterrupt):
+            await local_s3_fixture.abort_multipart_upload(
+                upload_id=upload_id,
+                bucket_id=CONFIG.inbox_bucket,
+                object_id=CONFIG.object_id,
+            )
+
+        event = event_schemas.FileUploadReceived(
+            file_id=CONFIG.object_id,
+            submitter_public_key=CONFIG.submitter_pubkey,
+            upload_date=datetime.utcnow().isoformat(),
+            decrypted_size=file_size,
+            expected_decrypted_sha256=checksum,
+        )
+        yield event
+
+
+@pytest_asyncio.fixture
+async def publish_and_record_fixture(
+    populated_bucket_fixture: event_schemas.FileUploadReceived,
+):
+    """Publish incoming event, record outgoing response event"""
+
+    async with EventRecorder(
+        kafka_servers=CONFIG.kafka_servers, topic="file_interrogation"
+    ) as event_recorder:
+        async with KafkaEventPublisher.construct(config=CONFIG) as publisher:
+            type_ = "file_upload_received"
+            key = CONFIG.object_id
+            topic = "file_uploads"
+            await publisher.publish(
+                payload=populated_bucket_fixture.dict(),
+                type_=type_,
+                key=key,
+                topic=topic,
+            )
```

Comment:
You cannot guarantee interrogation results will already be available when the event recorder stops.